settings/src/components/first-time/first-time.js

```diff
@@ -0,0 +1,212 @@
+						navigateToScreen( val );
+					} }
+				/>
+			),
+		},
+		totp: {
+			stepIndex: 1,
+			component: (
+				<TOTP
+					onSuccess={ () => {
+						navigateToScreen( 'backup-codes' );
+					} }
+				/>
+			),
+		},
+		webauthn: {
+			stepIndex: 1,
+			component: (
+				<WebAuthn
+					onKeyAdd={ () => {
+						navigateToScreen( 'backup-codes' );
+					} }
+				/>
+			),
+		},
+		'backup-codes': {
+			stepIndex: 2,
+			component: (
+				<BackupCodes
+					onSuccess={ () => {
+						navigateToScreen( 'congratulations' );
+					} }
+				/>
+			),
+		},
+		congratulations: {
+			stepIndex: 3,
+			component: (
+				<div className="">
+					<h3>Two-factor authentication setup is now complete! 🎉</h3>
+					<p>
+						To ensure the highest level of security for your account, please remember to
+						keep your authentication methods up-to-date. We recommend configuring
+						multiple authentication methods to guarantee you always have access to your
+						account.
+					</p>
+					<div className="wporg-2fa__submit-actions">
+						<Button
+							onClick={ () => {
+								const redirectTo = new URLSearchParams(
+									window.location.search
+								).get( 'redirect_to' );
+
+								if ( redirectTo && isValidUrl( redirectTo ) ) {
+									window.location.href = redirectTo;
+								} else {
+									window.location.href =
+										'//profiles.wordpress.org/me/profile/edit/group/1';
+								}
+							} }
+							isPrimary
+						>
+							Continue
+						</Button>
+					</div>
+				</div>
+			),
+		},
+	};
+
+	// Lock the scroll when the modal is open.
+	useEffect( () => {
+		const modal = modalRef.current;
+		const focusableElements = modal.querySelectorAll(
+			'a, button, input, textarea, select, [tabindex]:not([tabindex="-1"])'
+		);
+		const firstFocusableElement = focusableElements[ 0 ];
+		const lastFocusableElement = focusableElements[ focusableElements.length - 1 ];
+
+		const trapFocus = ( event ) => {
+			const isTabPressed = event.key === 'Tab' || event.keyCode === 9;
+			if ( ! isTabPressed ) {
+				return;
+			}
+
+			if ( event.shiftKey ) {
+				// eslint-disable-next-line @wordpress/no-global-active-element
+				if ( document.activeElement === firstFocusableElement ) {
+					lastFocusableElement.focus();
+					event.preventDefault();
+				}
+				return;
+			}
+
+			// eslint-disable-next-line @wordpress/no-global-active-element
+			if ( document.activeElement === lastFocusableElement ) {
+				firstFocusableElement.focus();
+				event.preventDefault();
+			}
+		};
+
+		modal.addEventListener( 'keydown', trapFocus );
+
+		document.querySelector( 'html' ).style.overflow = 'hidden';
+
+		// Focus the first focusable element in the modal when it opens
+		firstFocusableElement.focus();
+
+		return () => {
+			modal.removeEventListener( 'keydown', trapFocus );
+			document.querySelector( 'html' ).style.overflow = 'initial';
+		};
+	}, [ screen ] );
+
+	const currentStepIndex = screens[ screen ].stepIndex;
```

Comment:
We could do with a check here for the screen existing. Between switching from prod to sandbox I ended up on `?first-time=true&screen=account-status`, resulting in an error. Shouldn't be possible normally of course but it highlighted an edge case.